.travis.yml

```diff
@@ -1,20 +1,32 @@
 language: python
 cache: pip
-
-matrix:
+script: echo "Running tests against $(python --version) ..."
+jobs:
   include:
-    - python: 2.7
-    - python: 3.5
-    - python: 3.6
-    - python: 3.7
-    - python: 3.8-dev
-
-install:
-  - pip install -r requirements.txt
-  - pip install -U pytest pytest-cov
-
-script: pytest --cov tablib
-
-after_success:
-  - pip install -U codecov
-  - codecov
+  - stage: test
+    script: pytest --cov tablib
+    matrix:
+      include:
+      - python: 2.7
+      - python: 3.5
+      - python: 3.6
+      - python: 3.7
+      - python: 3.8-dev
+    install:
+    - pip install -r requirements.txt
+    - pip install -U pytest pytest-cov
+    after_success:
+    - pip install -U codecov
+    - codecov
+  - stage: deploy
+    script: skip
+    deploy:
+      provider: pypi
+      user: jazzband
```

Comment:
PyPI now supports uploading via API token:

https://pythoninsider.blogspot.com/2019/07/pypi-now-supports-uploading-via-api.html

> **Why:** These API tokens can **only** be used to upload packages to PyPI, and not to log in more generally. This makes it safer to automate package upload and store the credential in the cloud, since a thief who copies the token won't also gain the ability to delete the project, delete old releases, or add or remove collaborators. And, since the token is a long character string (with 32 bytes of entropy and a service identifier) that PyPI has securely generated on the server side, we vastly reduce the potential for credential reuse on other sites and for a bad actor to guess the token.

Something to consider?

Reply:
Not really since this does the release to the Jazzband package index (a staging areas of sorts) so that project leads can confirm each release. You'll get an emails once it's done. See https://jazzband.co/about/releases for more info.